### PR TITLE
feat(build): use jinja template for install script

### PR DESCRIPTION
**Pull request by aarnphm**
Signed-off-by: Aaron Pham <29749331+aarnphm@users.noreply.github.com>

## What does this PR address?

migrate install script to be in jinja for easier management

Fixes #(issue)

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, `pre-commit run -a` script has passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

File: src/bentoml/_internal/bento/build_config.py [hunk 1]
```diff
@@ -552,59 +563,15 @@ def write_to_bento(self, bento_fs: FS, build_ctx: str) -> None:
...
-        echo "WARNING: using BentoML version ${existing_bentoml_version}"
-    fi
-else
-    pip3 install bentoml=="$BENTOML_VERSION"
-fi
-"""
+            f.write(
+                self._jinja_environment.get_template("install.sh.j2").render(
+                    bentoml_version=clean_bentoml_version(BENTOML_VERSION),
+                    pip_args=" ".join(map(quote, args)),
```
Comment (frostming) [inline src/bentoml/_internal/bento/build_config.py:572]: shlex.join()

File: src/bentoml/_internal/bento/build_config.py [hunk 2]
```diff
@@ -511,6 +513,15 @@ def __attrs_post_init__(self):
     def is_empty(self) -> bool:
         return not self.requirements_txt and not self.packages
 
+    @functools.cached_property
+    def _jinja_environment(self) -> jinja2.Environment:
```
Comment (frostming) [inline src/bentoml/_internal/bento/build_config.py:517]: can we centralized the template loading, So we don't have to initialize jinja env in multiple places

**Comment by aarnphm** [comment]
@frostming should be good for another round of review here

File: src/bentoml/_internal/bento/build_config.py (hunk 2 above)
Comment (aarnphm) [reply src/bentoml/_internal/bento/build_config.py:517]: and also the environment here is different from the other templates.

Maybe we can move to a centralized format in a different PR?

**Review by frostming — COMMENTED** [review]
Mostly LGTM, just a minor concern that I suspect `[[ ]]` is more common in shell than `{{ }}`.

**Review by frostming — APPROVED** [review]

**Comment by aarnphm** [comment]
> Mostly LGTM, just a minor concern that I suspect `[[ ]]` is more common in shell than `{{ }}`.

Ah you are right, how about `<<` and `>>`?

**Pull request merged**